MaterialLib/PorousMedium/UnsaturatedProperty/CapillaryPressure/CreateCapillaryPressureModel.cpp

```diff
@@ -44,6 +44,16 @@ static std::unique_ptr<CapillaryPressureSaturation> createBrookCorey(
     //! \ogs_file_param{material__porous_medium__capillary_pressure__BrookCorey__sr}
     const double Sr = config.getConfigParameter<double>("sr");
 
+    double Sg_r = 0.0;
+    //! \ogs_file_param{material__porous_medium__capillary_pressure__BrookCorey__sg_res}
+    if (auto const Sg_res = config.getConfigParameterOptional<double>("sg_res"))
+    {
+        DBUG(
+            "Using value %g for nonwetting phase residual saturation in "
+            "capillary pressure model.",
+            (*Sg_res));
+        Sg_r = *Sg_res;
```

Comment:
actually Sg_res is a point. I change Sg_res to Sg_r_ptr. Others have been changed to Sg_r uniformly